src/reference/asciidoc/streams.adoc

```diff
@@ -121,6 +122,25 @@ private StreamsBuilderFactoryBean myKStreamBuilderFactoryBean;
 ----
 ====
 
+Starting with version 2.4.1, the factory bean has a new property `infrastructureCustomizer` with type `KafkaStreamsInfrastructureCustomizer`; this allows customization of the `StreamsBuilder` (e.g. to add a state store) and/or the `Topology` before the stream is created.
+
+====
+[source, java]
+----
+public interface KafkaStreamsInfrastructureCustomizer {
+
+	void configureBuilder(StreamsBuilder builder);
+
+	void configureTopology(Topology topology);
+
+}
+----
+====
+
+Default no-op implementations are provided to avoid having to implement both methods if one is not required.
+
+A `CompositeKafkaStreamsInfrastructureCustomizer` if you need multiple customizers.
```

Comment:
I know I'm not good in English, but feels like this sentence is missing a predicate.